src/state/markets/fifo.rs

```diff
@@ -750,9 +808,19 @@ impl<
                 order_packet.match_limit(),
                 base_lot_budget,
                 adjusted_quote_lot_budget,
+                order_packet.get_last_valid_slot().unwrap_or(0),
+                order_packet
+                    .get_last_valid_unix_timestamp_in_seconds()
+                    .unwrap_or(0),
             );
             let resting_order = self
-                .match_order(&mut inflight_order, trader_index, record_event_fn)
+                .match_order(
+                    &mut inflight_order,
+                    trader_index,
+                    record_event_fn,
+                    current_slot,
+                    current_unix_timestamp,
```

Comment:
weird to sometimes pass around the `get_clock_fn` and sometimes pass around the values, no?

Reply:
If `get_clock_fn` where an actual clock (ticking while executing) you really want the snapshot of that clock to be taken at the time of place. That makes passing its outputs into the private function `match_order` a reasonable decision.

Reply:
ya but it's not, and never will be